### ISSUE TITLE
link from comments

### ISSUE THREAD

**Issue by affeldt-aist** [affeldt-aist/coq2html#16]
* the ability to easily link to constants in the comments (by just surrounding them with `[` and `]`
(coqdoc feature not in coq2html)
@proux01

**Comment by yoshihiro503** [comment]
In `coqdoc`, expressions enclosed in `[]` are rendered as inline code but are not treated as links.
coq2html has the same feature.

As an additional feature not implemented in `coqdoc`, I think it would be useful to be able to implement links for code within comments, but this is difficult because that information is not included in the glob file. Since code within comments is not compiled, it may not only fail type checking but also contain syntactic errors.

Assuming the `rocq-lsp` server is enabled when generating HTML documentation, using the petanque protocol might make it possible to easily obtain `Locate` information for these code snippets within comments.

FYI: @gbdrt

**Comment by proux01** [comment]
Yes, this feature of coqdoc is convenient but pretty hackish. Not sure what's the best solution. `Locate` may not easily work as we often want to refer to things that will be fined after the comment so don't exist yet.

**Comment by yoshihiro503** [comment]
Thank you. You're probably right.

> Yes, this feature of coqdoc is convenient but pretty hackish. Not sure what's the best solution. `Locate` may not easily work as we often want to refer to things that will be fined after the comment so don't exist yet.
